feature_engine/encoding/woe.py

```diff
@@ -30,6 +30,44 @@
 from feature_engine.tags import _return_tags
 
 
+class WoE:
+    def _check_fit_input(self, X: pd.DataFrame, y: pd.Series):
+        """
+        Check that X is dataframe, and y a binary series with values 0 and 1.
+        """
+        X, y = check_X_y(X, y)
+
+        # check that y is binary
+        if y.nunique() != 2:
+            raise ValueError(
+                "This encoder is designed for binary classification. The target "
+                "used has more than 2 unique values."
+            )
+
+        # if target does not have values 0 and 1, we need to remap, to be able to
+        # compute the averages.
+        if y.min() != 0 or y.max() != 1:
+            y = pd.Series(np.where(y == y.min(), 0, 1))
+        return X, y
+
+    def _calculate_woe(self, X: pd.DataFrame, y: pd.Series, variable: str):
+        total_pos = y.sum()
+        inverse_y = y.ne(1).copy()
+        total_neg = inverse_y.sum()
+
+        pos = y.groupby(X[variable]).sum() / total_pos
+        neg = inverse_y.groupby(X[variable]).sum() / total_neg
+
```

Comment:
The same question about the groupby functionality used to calculate `pos`, but using `inverse_y`.

Proposed formula:

`neg = main.groupby(variable).sum()["inverse_y"] / total_neg`